### ISSUE TITLE
HDF5 data doesn't work for the Y_train single dimensional data

### ISSUE THREAD

**Issue by ashudeep** [fchollet/keras#279]
```
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
/data/src/net/regression_train.py in <module>()
     35 Y_train = HDF5Matrix("data/piece.h5", 'label', 0 , 40000000)
     36 Y_test = HDF5Matrix("data/piece.h5", 'label', 40000001 , 44899683)
---> 37 model.fit(X_train, Y_train, batch_size=1000, nb_epoch=1)
     38 
     39 score = model.evaluate(X_test, Y_test, batch_size=16)

/data/anaconda/lib/python2.7/site-packages/Keras-0.1.1-py2.7.egg/keras/models.pyc in fit(self, X, y, batch_size, nb_epoch, verbose, callbacks, validation_split, validation_data, shuffle, show_accuracy)
    164                     print("Train on %d samples, validate on %d samples" % (len(y), len(y_val)))
    165 
--> 166         index_array = np.arange(len(y))
    167 
    168         if verbose:

TypeError: len() of unsized object
```

@jfsantos added a helper class HDF5Matrix to use HDF5 data without loading it into the memory all at once. I received the above error when I tried to use it for my data. 

I figured out a few things:
- In my case Y_train is an array of the shape (N,) i.e. a single dimension
- len() of unsized object error is thrown because when models.fit calls standardize_y which actually adds another dimension to it by reshaping it to (len(N),1). But this creates a 'faulty' array which has no len attribute finally. 

Solution for me (still to check if it works): I will have to add my Ys into the dataset as arrays of size 1 rather than single numbers. This seems to break when using an HDF5Matrix helper class.

**Comment by ashudeep** [comment]
It seems to work for me after this one change in standardize_y function in models.py:
```python
def standardize_y(y):
    if not hasattr(y, 'shape'):
        y = np.asarray(y[:len(y)]) 
        #y = np.asarray(y)
    if len(y.shape) == 1:
        y = np.reshape(y, (len(y), 1))
    return y
```

**Comment by ashudeep** [comment]
But this doesn't seem to work when using validation_split option in models.fit( ) throwing the following error:
```
Traceback (most recent call last):
  File "regression_train.py", line 75, in <module>
    model.fit(X_train, y_train, batch_size=1024, nb_epoch=args.n, shuffle=False, callbacks=[history, ch
eckpointer], validation_split=0.1)
  File "build/bdist.linux-x86_64/egg/keras/models.py", line 162, in fit
  File "build/bdist.linux-x86_64/egg/keras/models.py", line 38, in slice_X
  File "build/bdist.linux-x86_64/egg/keras/utils/io_utils.py", line 48, in __getitem__
  File "h5py/_objects.pyx", line 54, in h5py._objects.with_phil.wrapper (-------src-dir-------/h5py-2.4
.0/h5py/_objects.c:2405)
  File "h5py/_objects.pyx", line 55, in h5py._objects.with_phil.wrapper (-------src-dir-------/h5py-2.4
.0/h5py/_objects.c:2362)
  File "/data/anaconda/lib/python2.7/site-packages/h5py/_hl/dataset.py", line 456, in __getitem__
    arr = numpy.ndarray(mshape, new_dtype, order='C')
MemoryError
```
@jfsantos If there is an immediate solution to the bug(s), can you please help. I will have time to look into this later only. I can work without a validation set meanwhile!

**Comment by fchollet** [comment]
> `MemoryError`

You ran out of RAM. How is that a Keras-related error?

**Comment by ashudeep** [comment]
The traceback in the first case is not due to a memory error. I simply fixed it by the edit mentioned in message#2. 
But the edit doesn't let me use validation_split option (the one where it seems to runs out of Memory). But, I am not sure if it really runs out of memory as I was monitoring memory usage while it crashed and also tried with much smaller batch sizes. I am quite sure it wasn't the memory that ran out. I will recheck though. But anyways, I had to make edit to ```standardize_y''' in the first place.
